### DISCUSSION TITLE
Not able to add lineItem in a checkout

### DISCUSSION THREAD

**Discussion by rajatabcx** [Shopify/storefront-api-feedback#235] [Help] [answered]
I have some webhooks(checkout/create, checkout/update) in place which is storing the checkoutToken, checkoutId and abandonedCheckoutURL, but when i am using the storefront api to add a new lineItem to checkout its showing that the checkout does not exist, but i am able to fetch the checkout data via admin api

even if i add the line item in cart using the cartToken (also got from the webhook), the mutation succeeded, but the change is not reflected in the checkout.

## Storefront mutation

```js
      const variantId = btoa(`gid://shopify/ProductVariant/${data.variantId}`);
      const checkoutId = btoa(
        `gid://shopify/Checkout/${checkout.shopifyCheckoutId}?key=${key}`,
      );

      const {
        data: resData,
        errors,
        extensions,
      } = await shopify.request(
        `mutation {
          checkoutLineItemsAdd(
            checkoutId: "${checkoutId}",
            lineItems: [
              {
                variantId:"${variantId}",
                quantity: 1
              }
            ]
          ) {
            checkout {
              id
            }
            checkoutUserErrors {
              code
              field
              message
            }
          }
        }
      `,
      );
```

## error

```js
[
  {
    code: 'INVALID',
    field: [ 'checkoutId' ],
    message: 'Checkout does not exist'
  }
]
```

**Comment by vixdug** [comment, marked as answer]
Hi @rajatabcx - the API is only able to modify checkouts that are created via the application itself. Please note these mutations are [deprecated and replaced by the Cart mutations](https://shopify.dev/changelog/deprecation-of-checkout-apis).
